doc/spec/namespace.md

```diff
@@ -0,0 +1,144 @@
+
+`<fully qualified name> -> <domain>/<path>`
+
+#### HTTPS
+The discovery related metadata will be fetched via HTTPS from the DNS resolved
+location using the remaining namespace path elements as the HTTP path in a GET
+request. A discovery request url would be in the format
+`https://<domain>/<name>?docker-discovery=1`
+
+For example, “example.com/foo/bar” would create a url
+“https://example.com/foo/bar?docker-discovery=1”
+
+##### HTML Response
+~~~html
+<meta name="docker-scope" content="example.com">
+<meta name=“docker-registry” content=“push,pull v2 https://registry.example.com/v2/”>
+<meta name=“docker-registry” content=“push,pull v1 https://registry.example.com/v1/”>
+<meta name=“docker-registry” content=“pull v2 https://registry.mirror.com/v2/”>
+<meta name=“docker-registry” content=“pull,notag v2 http://registry.mirror.com/v2/”>
+<meta name=“docker-index” content=“v1 https://search.mirror.com/v1/”>
+<meta name=“docker-certificate” content=“https://trust.example.com/registry.cert}”>
+~~~
+
+#### Fallback (Compatibility)
+If HTTPS is not implemented for a namespace, a fallback protocol may be
+used when the namespace prefix is explicitly marked as insecure. The
+fallback process involves attempting to ping possible registry api endpoints to
+determine the set of endpoints and using no trust model.
+
+### Extensibility
+A custom method may be used to provide discovery by implementing the
+`NamespaceResolver` interface.
+
+### Scope
+The namespace information produced from the discovery process may contain a scope
+field. The scope field means the information may apply to any namespace with a
+prefix of the given scope. The Scope prefix will always be applied with a path
+separator. If the scope field is ommitted, the information may not be applied to
+any other namespace.
+
+### Endpoints
+The registry api endpoints each contain a version (may be v1 or v2 registries) and
+may either be pull only mirrors or full registries. The trust model applies to
+each registry api endpoint and may not be overloaded by an individual endpoint. The
+trust model defines the method for verifying the content retrieved from an
+endpoint, not the method for authentication or authorization.  Each registry api
+endpoint is responsible for specifying its authentication or authorization
+method. 
+
+It may also be possible in the future to extend these endpoints to support
+direct downloads of tarballs, such as the result from a `docker save`.
+
+## Certificates
+Each namespace scope may have a certificate which can be used to sign keys for each
+individual repository. The certificate must be scoped to the namespace and be used
+to sign certificates using the fully qualified name for each repository. If
+namespace does not contain a certificate, the namespace will be considered
+untrusted for any repositories it contains. It is up to the client whether to
+require these certificates to be signed by a root CA.
```

Comment:
x509 is the only thing being discussed right now.  I am intentionally not specifying right here because the trust proposal is still being drafted with the specifics. I will likely track those changes then point there later.